### PR TITLE
feat(EMS-1921):  custom domain names for Azure front door

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction :pencil2:
Ensure custom domains are created during IaC execution.
⚠️ Please not `domains` job will fail unless SSLs are manually added to KV which has not been covered as part of this PR.

## Resolution :heavy_check_mark:
* Managed identity creation at FD level.
* Ensure Key Vault policy to allow correct permissions for FD to fetch respective certificates.
* Introduction of a new domains job responsible for:
  * Add custom certificates from KV to FD secrets
  * Creation of custom domains with FD secrets.
  * Association of endpoints with respective custom domains.

## Miscellaneous :heavy_plus_sign:
* Added values to `.env.sample` for API container.

### CODE REVIEW

File: .github/workflows/infrastructure.yml [hunk 1]
```diff
@@ -564,6 +564,26 @@ jobs:
             --name IPAllowListRule \
             --policy-name waf${{ env.PRODUCT }}${{ env.TARGET }}${{ vars.VERSION }}
 
+      - name: Identity 🪪
```
Comment (Zainzzkk) [inline .github/workflows/infrastructure.yml:567]: emoji not showing for me, or should it be a ?

File: .github/workflows/infrastructure.yml [hunk 2]
```diff
@@ -1058,3 +1078,84 @@ jobs:
...
+            --use-latest-version true \
+            --secret-source $(az keyvault list --query '[].id' -o tsv)/secrets/kv-certificate-quote-latest
+
+            az afd secret create \
+            --name kv-certificate-apply-latest \
+            --profile-name frontdoor-${{ env.PRODUCT }}-${{ env.TARGET }}-${{ vars.VERSION }} \
+            --use-latest-version true \
+            --secret-source $(az keyvault list --query '[].id' -o tsv)/secrets/kv-certificate-apply-latest
+
+      - name: Domains 🌐
```
Comment (Zainzzkk) [inline .github/workflows/infrastructure.yml:1131]: we have 2 Domains as name, could one be a different name?

File: .github/workflows/infrastructure.yml (hunk 1 above)
Comment (abhi-markan) [reply .github/workflows/infrastructure.yml]: Have updated.

File: .github/workflows/infrastructure.yml (hunk 2 above)
Comment (abhi-markan) [reply .github/workflows/infrastructure.yml]: Have updated.

**Review by Zainzzkk — APPROVED** [review]

**Pull request merged**